staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -291,7 +293,7 @@ func (s preparedGenericAPIServer) NonBlockingRun(stopCh <-chan struct{}) error {
 	// Start the audit backend before any request comes in. This means we cannot turn it into a
 	// post start hook because without calling Backend.Run the Backend.ProcessEvents call might block.
 	if s.AuditBackend != nil {
-		if err := s.AuditBackend.Run(stopCh); err != nil {
```

Comment:
why doesn't stopCh work here?

Reply:
the stopCh close may before apiserver stop accept new request, so new request audit events are dropped.

Reply:
Can you rebase onto the PreShutdownHook PR? This func will change I think.

Reply:
yes, I will first rebase then modify for your comments